lib/widgets/compose_box.dart

```diff
@@ -927,38 +988,22 @@ class _SendButtonState extends State<_SendButton> {
 
   @override
   Widget build(BuildContext context) {
-    final disabled = _hasValidationErrors;
-    final colorScheme = Theme.of(context).colorScheme;
+    final designVariables = DesignVariables.of(context);
     final zulipLocalizations = ZulipLocalizations.of(context);
 
-    // Copy FilledButton defaults (_FilledButtonDefaultsM3.backgroundColor)
-    final backgroundColor = disabled
-      ? colorScheme.onSurface.withValues(alpha: 0.12)
-      : colorScheme.primary;
-
-    // Copy FilledButton defaults (_FilledButtonDefaultsM3.foregroundColor)
-    final foregroundColor = disabled
-      ? colorScheme.onSurface.withValues(alpha: 0.38)
-      : colorScheme.onPrimary;
+    final iconColor = _hasValidationErrors
+      ? designVariables.icon.withFadedAlpha(0.5)
+      : designVariables.icon;
 
-    return Ink(
-      decoration: BoxDecoration(
-        borderRadius: const BorderRadius.all(Radius.circular(8.0)),
-        color: backgroundColor,
-      ),
+    return SizedBox(
+      width: _composeButtonWidth,
       child: IconButton(
         tooltip: zulipLocalizations.composeBoxSendTooltip,
-        style: const ButtonStyle(
-          // Match the height of the content input.
-          minimumSize: WidgetStatePropertyAll(Size.square(_sendButtonSize)),
-          // With the default of [MaterialTapTargetSize.padded], not just the
-          // tap target but the visual button would get padded to 48px square.
-          // It would be nice if the tap target extended invisibly out from the
-          // button, to make a 48px square, but that's not the behavior we get.
-          tapTargetSize: MaterialTapTargetSize.shrinkWrap,
-        ),
-        color: foregroundColor,
-        icon: const Icon(Icons.send),
+        icon: Icon(ZulipIcons.send,
+          // We set the color on `Icon` instead of `IconButton` because the
+          // Figma design does not apply it on the background of the button
+          // for states like highlighted, etc.
+          color: iconColor),
```

Comment:
I don't follow this comment. This `color` property is about the foreground color, not the background, right?

Reply:
We do want to set the foreground color to `iconColor`; either `IconButton.color` or `Icon.color` will work.

We do want to avoid the default background colors (i.e. the color of the button: `IconButton.hoverColor`, `IconButton.highlightColor`); these are derived from `IconButton.color` if it is set (it is the fallback when the ambient `ButtonStyle.overlayColor` is not set, see [`IconButton.styleFrom`](https://github.com/flutter/flutter/blob/eaa99f28b1cebf39be5fe92633d79af0b64fe329/packages/flutter/lib/src/material/icon_button.dart#L646-L659), which is called by `IconButton.build`).

This exact behavior of falling back to the foreground color (`IconButton.color`) doesn't seem to be mentioned on the colors' dartdoc. When we set `IconButton.highlightColor` through an `IconButtonTheme`, it gets overridden by the color derived from `IconButton.color`.

we can clarify this bit by mentioning the specific button styles involved.